### PR TITLE
Add tests for extract tag functions in prereleaselifecyclegenerators

### LINKED ISSUE #119289: Improve test coverage on code generator
The commands under https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/code-generator/cmd 
will benefit of a batter unit test coverage as seen in 

_Originally posted by @aojea in https://github.com/kubernetes/kubernetes/pull/119268#discussion_r1261752502_

### PR DESCRIPTION
**Pull request by DamianSawicki**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Adds unit tests for `extractTag`, `extractEnabledTypeTag`, and `extractReplacementTag` from the file `code-generator/cmd/prerelease-lifecycle-gen/prerelease-lifecycle-generators/status.go` as requested in issue https://github.com/kubernetes/kubernetes/issues/119289.

#### Which issue(s) this PR fixes:

Fixes #119289

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/code-generator/cmd/prerelease-lifecycle-gen/prerelease-lifecycle-generators/status_test.go [hunk 1]
```diff
@@ -146,3 +147,293 @@ func Test_extractKubeVersionTag(t *testing.T) {
...
+				name:         "duplicated variable",
+				tagComments:  comments,
+				variableName: "variable",
+				wantValue:    nil, // TODO: Today it calls klog.Fatal, check how to capture it
+			},
+			{
+				name:         "variable without marker test",
+				tagComments:  comments,
+				variableName: "multi-valuedVariable",
+				wantValue:    nil, // TODO: Today it calls klog.Fatal, check how to capture it
```
Comment (aojea) [inline staging/src/k8s.io/code-generator/cmd/prerelease-lifecycle-gen/prerelease-lifecycle-generators/status_test.go:205]: @pohly didn't we have some klog-testing feature to test this?

Comment (pohly) [reply staging/src/k8s.io/code-generator/cmd/prerelease-lifecycle-gen/prerelease-lifecycle-generators/status_test.go:205]: You remember correctly. `klog.OsExit` gets exported and can be overridden:

https://github.com/kubernetes/klog/blob/6632ba5cc9a5331c853a2c6dbc0fbb125e6e7b94/exit.go#L38-L43

Comment (DamianSawicki) [reply staging/src/k8s.io/code-generator/cmd/prerelease-lifecycle-gen/prerelease-lifecycle-generators/status_test.go]: Thanks @aojea and @pohly for the hint! I used it in the newly added test and also in the the pre-existing one.

File: staging/src/k8s.io/code-generator/cmd/prerelease-lifecycle-gen/prerelease-lifecycle-generators/status_test.go [hunk 2]
```diff
@@ -32,6 +35,12 @@ var mockType = &types.Type{
 }
 
 func Test_extractKubeVersionTag(t *testing.T) {
+	oldKlogOsExit := klog.OsExit
+	defer func() {
+		klog.OsExit = oldKlogOsExit
+	}()
+	klog.OsExit = customExit
```
Comment (aojea) [inline staging/src/k8s.io/code-generator/cmd/prerelease-lifecycle-gen/prerelease-lifecycle-generators/status_test.go:42]: @pohly do you mind reviewing these test changes are ok?

**Comment by aojea** [comment]
/assign @sttts 

as owner

```

**Comment by rikatz** [comment]
Overall this is great and lgtm!!

If you want to copy a test for [isApiType](https://github.com/rikatz/kubepug/commit/a6518c096871590ae10f639fb25a64520a234f83#diff-46d1046bc155d61aaa69c14e440a817bfc4cb44864008b6f3c2e4831d13aa89eR300) feel free to copy from [here](https://github.com/rikatz/kubepug/blob/main/generator/deprecations/gen_test.go#L300).

Once this gets merged I will also crossport back to kubepug :) Thanks for helping on increasing the coverage test!
/lgtm

**Comment by DamianSawicki** [comment]
> If you want to copy a test for [isApiType](https://github.com/rikatz/kubepug/commit/a6518c096871590ae10f639fb25a64520a234f83#diff-46d1046bc155d61aaa69c14e440a817bfc4cb44864008b6f3c2e4831d13aa89eR300) feel free to copy from [here](https://github.com/rikatz/kubepug/blob/main/generator/deprecations/gen_test.go#L300).

Thanks for pointing it out, copied!

> Once this gets merged I will also crossport back to kubepug :) Thanks for helping on increasing the coverage test! /lgtm

I'm very glad I could help a bit :)

**Comment by DamianSawicki** [comment]
@rikatz your lgtm disappeared when I pushed `Test_isAPIType`. Could you please have a look again and lgtm?

**Comment by DamianSawicki** [comment]
Hey @thockin, could you please move it forward?

**Comment by DamianSawicki** [comment]
Ping @thockin.

**Pull request merged**